nursery_store.go

```diff
@@ -0,0 +1,1489 @@
+		if err = ns.forEachHeightPrefix(tx, cribPrefix, height,
+			func(buf []byte) error {
+
+				// We will attempt to deserialize all outputs
+				// stored with the crib prefix into babyOutputs,
+				// since this is the expected type that would
+				// have been serialized previously.
+				var baby babyOutput
+				babyReader := bytes.NewReader(buf)
+				if err := baby.Decode(babyReader); err != nil {
+					return err
+				}
+
+				babies = append(babies, baby)
+
+				return nil
+
+			}); err != nil {
+			return err
+		}
+
+		// Append each kindergarten output to our list of kidOutputs.
+		return ns.forEachHeightPrefix(tx, kndrPrefix, height,
+			func(buf []byte) error {
+				// We will attempt to deserialize all outputs
+				// stored with the kindergarten prefix into
+				// kidOutputs, since this is the expected type
+				// that would have been serialized previously.
+				var kid kidOutput
+				kidReader := bytes.NewReader(buf)
+				if err := kid.Decode(kidReader); err != nil {
+					return err
+				}
+
+				kids = append(kids, kid)
+
+				return nil
+
+			})
+
+	}); err != nil {
+		return nil, nil, nil, err
+	}
+
+	return finalTx, kids, babies, nil
+}
+
+// FetchPreschools returns a list of all outputs currently stored in the
+// preschool bucket.
+func (ns *nurseryStore) FetchPreschools() ([]kidOutput, error) {
+	var kids []kidOutput
+	if err := ns.db.View(func(tx *bolt.Tx) error {
+
+		// Retrieve the existing chain bucket for this nursery store.
+		chainBucket := tx.Bucket(ns.pfxChainKey)
+		if chainBucket == nil {
+			return nil
+		}
+
+		// Load the existing channel index from the chain bucket.
+		chanIndex := chainBucket.Bucket(channelIndexKey)
+		if chanIndex == nil {
+			return nil
+		}
+
+		// Construct a list of all channels in the channel index that
+		// are currently being tracked by the nursery store.
+		var activeChannels [][]byte
+		if err := chanIndex.ForEach(func(chanBytes, _ []byte) error {
+			activeChannels = append(activeChannels, chanBytes)
+			return nil
+		}); err != nil {
+			return err
+		}
+
+		// Iterate over all of the accumulated channels, and do a prefix
+		// scan inside of each channel bucket. Each output found that
+		// has a preschool prefix will be deserialized into a kidOutput,
+		// and added to our list of preschool outputs to return to the
+		// caller.
+		for _, chanBytes := range activeChannels {
+			// Retrieve the channel bucket associated with this
+			// channel.
+			chanBucket := chanIndex.Bucket(chanBytes)
+			if chanBucket == nil {
+				continue
+			}
+
+			// All of the outputs of interest will start with the
+			// "pscl" prefix. So, we will perform a prefix scan of
+			// the channel bucket to efficiently enumerate all the
+			// desired outputs.
+			c := chanBucket.Cursor()
+
+			// Seek and iterate over all outputs starting with the
+			// prefix "pscl".
+			pfxOutputKey, kidBytes := c.Seek(psclPrefix)
+			for bytes.HasPrefix(pfxOutputKey, psclPrefix) {
```

Comment:
I broke it up this way because go fmt didn't give a lot of room in how to break the lines, and just thought it was more important to have descriptive identifiers. I tried a number of combinations but couldn't find a oneliner that was satisfactory. Open to more suggestions though 🙂

Reply:
Shorter variable names? 😛 
```go
for k, v := c.Seek(psclPrefix); bytes.HasPrefix(k, psclPrefix); k, v = c.Next() {

}
```